### ISSUE TITLE
SecureSocketImpl::currentSession() always return null

### ISSUE THREAD

**Issue by micheleselea** [pocoproject/poco#4252]
In version 1.12.5p1 I can't have FTPS connection working anymore, because FTPS Dataconnection works only reusing the session of command connection.
It seems that calling `currentSession()` on the established command secure socket return always null, because `SecureSocketImpl::onSessionCreated` is never called even if session caching is enabled in Server Context using
```
enableSessionCache(true, "some id");
setSessionTimeout(timeout);
setSessionCacheSize(SSL_SESSION_CACHE_MAX_SIZE_DEFAULT);
```

**Comment by micheleselea** [comment]
I found that currentSession() server side works only if client side is enabled enableSessionCache
The problem is that FileZilla client with FTPS Explicit mode do not set that. But I think something is missing because I should be able to get currentSession even if client do not set that session cache, Am I wrong?

**Comment by micheleselea** [comment]
Moreover I think we have to think about the function
```
void Context::disableStatelessSessionResumption()
{
#if defined(SSL_OP_NO_TICKET)
	SSL_CTX_set_options(_pSSLContext, SSL_OP_NO_TICKET);
#endif
}
```
and the acceptSSL that now is like
```
void SecureSocketImpl::acceptSSL()
{
	poco_assert (!_pSSL);

	BIO* pBIO = BIO_new(BIO_s_socket());
	if (!pBIO) throw SSLException("Cannot create BIO object");
	BIO_set_fd(pBIO, static_cast<int>(_pSocket->sockfd()), BIO_NOCLOSE);

	_pSSL = SSL_new(_pContext->sslContext());
	if (!_pSSL)
	{
		BIO_free(pBIO);
		throw SSLException("Cannot create SSL object");
	}

#if OPENSSL_VERSION_NUMBER >= 0x1010100fL
	/* TLS 1.3 server sends session tickets after a handhake as part of
	* the SSL_accept(). If a client finishes all its job before server
	* sends the tickets, SSL_accept() fails with EPIPE errno. Since we
	* are not interested in a session resumption, we can not to send the
	* tickets. */
	if (1 != SSL_set_num_tickets(_pSSL, 0))
	{
		BIO_free(pBIO);
		throw SSLException("Cannot create SSL object");
	}
	//Otherwise we can perform two-way shutdown. Client must call SSL_read() before the final SSL_shutdown().
#endif

	SSL_set_bio(_pSSL, pBIO, pBIO);
	SSL_set_accept_state(_pSSL);
	SSL_set_ex_data(_pSSL, SSLManager::instance().socketIndex(), this);
	_needHandshake = true;
}
```
I think the ifdef code in acceptSSL is to disable session resumption but probably we have to think if it's still needed I think we put that years ago when Openssl 3.0 first appear, probably it's not needed anymore, because reading the comment can block session resumption. I already tried to check if currentSession works without that code, but even without the SSL_set_num_tickets the function is not working: I manage to get it working again only restoring preview code or connecting with a ssl client explicit enabling session resumption. But as I told before I think is not a correct behavior

**Comment by aleks-f** [comment]
send a pull request, with explanation and a test case if you want this addressed for 1.13

**Comment by aleks-f** [comment]
#4103
